ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/layered/AbstractTransactionsLayer.java

```diff
@@ -326,11 +327,19 @@ protected TransactionAddedResult addToNextLayer(
       final PendingTransaction pendingTransaction,
       final int distance,
       final AddReason addReason) {
-    final int nextLayerDistance;
+    int nextLayerDistance;
     if (senderTxs.isEmpty()) {
       nextLayerDistance = distance;
     } else {
-      nextLayerDistance = (int) (pendingTransaction.getNonce() - (senderTxs.lastKey() + 1));
+      try {
+        nextLayerDistance = (int) (pendingTransaction.getNonce() - (senderTxs.lastKey() + 1));
+      } catch (NoSuchElementException nse) {
+        // There is a timing condition where isEmpty() returns false, then another thread removes
+        // the last key before our call to lastKey(). There's no benefit to us adding a mutex to
+        // prevent the other thread updating the map so we just catch the exception and treat the
+        // map as empty.
+        nextLayerDistance = distance;
+      }
```

Comment:
this change is not needed here since each layer is externally synchronized and thread safe

Reply:
OK thanks for confirming, I've removed that change